DFE-Digital/sts-plan-technology-for-your-school
# feat: Record user signins with no organisation

## Overview

Addresses ticket [#229208](https://dfe-ssp.visualstudio.com/s190-schools-technology-services/_workitems/edit/229208)

Records sign-ins in the database for users without an organisation

## Changes

- Rather than changing the `RecordUserSignInDTO` which is used in a fair few places, this instead makes methods like `CreateUser` require the user ID only
- Adds a new `RecordSignInUserOnly` for a sign in without an organisation that doesn't add claims for using the site but makes a database call to store the attempt
- Makes establishment column in the db nullable to account for this change

## How to review the PR

- When running locally, create a new DSI sign in account https://dev-services.signin.education.gov.uk/
  - Note that the create new account on the dev start page links to dsi prod so you need to use the link in this PR
- Use your email  +test (e.g. katie.gardner+test@...) 
- Don't request to be part of an organisation when creating your account
- Try to log into plan tech, you should get the no organisation error page
- Check the database, you should have a new entry with null establishment 
  - I used a local test db, if you want to use the dev one, run the SQL script against it first to make the column nullable

## Checklist
